src/Sarif/Core/Result.cs

```diff
@@ -10,5 +12,63 @@ public partial class Result
         public bool ShouldSerializeWorkItemUris() { return this.WorkItemUris != null && this.WorkItemUris.Any((s) => s != null); }
 
         public bool ShouldSerializeLevel() { return this.Level != FailureLevel.Warning; }
+
+        /// <summary>
+        ///  Look up the ReportingDescriptor for this Result.
+        /// </summary>
+        /// <param name="run">Run instance containing this Result</param>
+        /// <returns>ReportingDescriptor for Result Rule, if available</returns>
+        public ReportingDescriptor GetReportingDescriptor(Run run)
```

Comment:
>GetReportingDescriptor [](start = 35, length = 22)

The method name is ambiguous since it could refer to the Rule or the Taxa. Perhaps it should be GetRule().

Reply:
Good idea - renamed.